cbqueryx/ensureindexhelper.go

```diff
@@ -0,0 +1,172 @@
+package cbqueryx
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/couchbase/gocbcorex/cbhttpx"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+)
+
+type EnsureIndexHelper struct {
+	Logger     *zap.Logger
+	UserAgent  string
+	OnBehalfOf *cbhttpx.OnBehalfOfInfo
+
+	BucketName     string
+	ScopeName      string
+	CollectionName string
+	IndexName      string
+
+	confirmedEndpoints []string
+}
+
+type NodeTarget struct {
+	Endpoint string
+	Username string
+	Password string
+}
+
+type EnsureIndexPollOptions struct {
+	Transport http.RoundTripper
+	Targets   []NodeTarget
+}
+
+func (e *EnsureIndexHelper) PollCreated(ctx context.Context, opts *EnsureIndexPollOptions) (bool, error) {
+	return e.pollAll(ctx, opts, func(numRows int) bool {
+		// If there are rows then the endpoint knows the index.
+		if numRows == 0 {
+			e.Logger.Debug("target responded with success, but the target still didn't have the index")
+			return false
+		} else if numRows > 1 {
+			e.Logger.Debug("query returned %d rows rather than 1", zap.Int("num-rows", numRows))
```

Comment:
What does it mean if we have more than one row?

Reply:
That the impossible has happened. 2 indexes with the same name on the same collection.